### ISSUE TITLE
[Feature] Give randomized main settings a default chance to counter odds from high optional choice counts

### ISSUE THREAD

**Issue by Kongolan** [TestRunnerSRL/OoT-Randomizer#851]
So after testing the randomized main rule option for a little bit, i came to notice the obvious. Almost all the time, there is a MQ dungeon (12 options) in the pool or any entrance shuffled (7 choices) just because the acutal count of choices is very high.

So my suggestion is to have a default feature set, by what the items are set to when greyed out. Above on the right next to the **"randomize main rule" setting there will be a checkbox where you can put in the percentage for the "default" rule to be chooses (from 1-100%).** So having 60% of the default setting, you at least can make sure, that stuff like master quest dungeons are not always in, while still having the same feel of not knowing if skulls are shuffled or other stuff is basically random, which you will not know like removed boss keys.

For me this way is the best idea to have the randomized main rule setting to become useful for those, who mainly play accessible rulesets and might want some settings to occur rarely.

While i already thought about this, we could also have a **checkbox behind every setting (only when the randomize main rule setting is on), which includes this option in the randomization**. So we can simply check MQ dungeons / entrance shuffle / or keysanity out, if we want to, while still having the other stuff randomized fully.

Another thing, where this checkbox can be handy is to also add a randomize other settings as well, where some people might want to always have the fast chest cutscenes or fast travel, but other stuff like free scarescrow song randomized.

So basically my suggestions are:
1. percentage for default option (which is the greyed out option chooses before we click randomize main rules), which can be like the accessible preset and still make sure that some high count options are **always** on
2. check boxes behind every option to include it into the randomization of the rule
3. randomize other settings category as well

Since i have only a little knowledge with python it would also help me if anyone can hint me, where the randomize main rule GUI setting is created, so i can check how it is called and which python file actually does the randomization for it. Since i checked it out already, but i couldn't find it on the new electron GUI.

**Comment by TestRunnerSRL** [comment]
The options actually do have weights. for example entrance shuffle has a 50% chance to be off, then an equal 1/8 between the other options. Some options like MQ just always set the random option to be random, but I can adjust any distributions if need be. That being said I'm definitely not adding a ton of check boxes and such to the gui for every option.
